decoder/src/log/json_logger.rs

```diff
@@ -68,30 +65,27 @@ impl JsonLogger {
     }
 }
 
-fn extract_path(
-    module_path: Option<&str>,
-) -> (
-    Option<String>,
-    Option<Vec<String>>,
-    Option<String>,
-    Option<bool>,
-) {
-    let module_path = if let Some(a) = module_path {
-        a.split("::").collect::<Vec<_>>()
-    } else {
-        return (None, None, None, None);
-    };
-
-    let idx = module_path.len() - 1;
+#[derive(Serialize)]
+struct Path {
```

Comment:
I would suggest naming this `ModulePath` so that it's not mistaken with `std::path::Path`